src/index.js

```diff
@@ -165,5 +180,71 @@ export default class ScrollPlugin extends Plugin {
                 swup.scrollTo(0, this.shouldAnimate('betweenPages'));
             }
         }
-    };
+    }
+
+    onWillReplaceContent = () => {
+        this.storeScrollPositions(this.previousUrl);
+        this.previousUrl = window.location.href;
+    }
+
+    onClickLink = (e) => {
+        this.deleteStoredScrollPositions(e.delegateTarget.href);
```

Comment:
Why do we have to delete the scroll position on link click?

Reply:
Automatically Restoring the scroll position should only be done while navigating the history. Otherwise, it should be 0, just like the browser does it. Maybe we could introduce another attribute here, something like `[data-swup-history-link]` or the like. Then we could preserve the scroll positions for links matching this attribute.

Reply:
The link would then basically behave for scroll positions like a link with `href="javascript:history.back()"`.

Reply:
Ah yes, makes sense.

Reply:
I don't think we need an attr for changing the behavior, though, it should be fine the way it is.

Reply:
I just remembered a few cases I had in the past, where clients wanted to go back to a list of posts from a detail page, and the browser back button wasn't enough for them. So we *do* need a solution for that IMO. This is what I came up with: [d10f95d](https://github.com/swup/scroll-plugin/pull/26/commits/d10f95d242eb697bc82dba379f25e11ae0136377)

The new option behaves just like swup's own option `skipPopstateHandling` and could be used like this, for example:

```js
{
  skipDeletingScrollPositions: el => {
    return el.classList.contains('backlink')
  }
}
```